url0e7d57473a_kyorohiro_HetimaUtil_tgz-pJ8-net_hetimatan_net_http_HttpServer3xxJ8/src/net/hetimatan/util/http/HttpRequestHeader.java

```diff
@@ -0,0 +1,85 @@
+package net.hetimatan.util.http;
+
+
+import java.io.IOException;
+import java.io.OutputStream;
+
+import net.hetimatan.io.file.MarkableFileReader;
+import net.hetimatan.io.file.MarkableReader;
+import net.hetimatan.io.file.MarkableReaderHelper;
+import net.hetimatan.io.filen.CashKyoroFile;
+
+// KEY ":" VALUE CRLF
+public class HttpRequestHeader extends HttpObject {
+
+	private String mKey = "";
+	private String mValue = "";
+	public static final String SEPARATOR = ":";
+	public static final String HEADER_CONTENT_LENGTH = "Content-Length";
+	public static final String HEADER_CONTENT_TyPE = "Content-Type";
+	public static final String HEADER_LOCATION = "Location";
+	public static final String HEADER_HOST = "Host";
+
+	public HttpRequestHeader(String key, String value) {
+		mKey = key;
+		mValue = value;
+	}
+
+	@Override
+	public void encode(OutputStream output) throws IOException {
+		output.write(mKey.getBytes());
+		output.write(SEPARATOR.getBytes());
+		output.write(" ".getBytes());
+		output.write(mValue.getBytes());
+		output.write(CRLF.getBytes());
+	}
+
+	public String getKey() {
+		return mKey;
+	}
+
+	public String getValue() {
+		return mValue;
+	}
+
+	public static HttpRequestHeader decode(String path) throws IOException {
+		CashKyoroFile vFile = new CashKyoroFile(512, 2);
+		vFile.addChunk(path.getBytes());
+		MarkableReader reader = new MarkableFileReader(vFile, 256);
+		return HttpRequestHeader.decode(reader);
+	}
+
+	public static HttpRequestHeader decode(MarkableReader reader) throws IOException {
+		String key = _key(reader);
+		_spa(reader); _sep(reader);_spa(reader);
+		String value = _val(reader);
+		_crlf(reader);
+		return new HttpRequestHeader(key, value);
+	}
+
+	private static String _key(MarkableReader reader) throws IOException {
+		try {
+			return new String(MarkableReaderHelper.asciiAndGet(reader,":".getBytes(), Integer.MAX_VALUE));
+		} catch(IOException e) {
+			throw e;
+		}
+	}
+
+	private static String _val(MarkableReader reader) throws IOException {
+		try {
+			return new String(MarkableReaderHelper.asciiAndGet(reader, "".getBytes(), Integer.MAX_VALUE));
+		} catch(IOException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An exception is caught and re-thrown within the catch clause without any other operations being performed on the exception, making the catch clause unnecessary.

**Fix**: Perform additional operations on the exception that has been caught. You can wrap a standard exception into a custom exception to add additional information or special handling. When you do that, make sure to set the original exception as the cause. Otherwise, you will lose the stack trace and message of the original exception, which will make it difficult to analyze the event that caused the exception.  
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)